### PR TITLE
(configurator) fix metadata + extract configurators in own component

### PR DESCRIPTION
**Pull request by hebus**
* for each page, configurators are now in their own component
* fix metadata changes (just to fix the CI build)

### CODE REVIEW

File: projects/vanilla/src/app/configurators/metadata-configurator.component.ts [hunk 1]
```diff
@@ -25,11 +26,15 @@ import { ComponentConfig, ConfiguratorContext } from "@sinequa/ngx-ui-builder";
...
-  @Input() metadata: string[];
+  @Input() metadata: MetadataConfig[];
 
   isEntity: boolean;
 
-  constructor(public appService: AppService) { }
+  constructor(public appService: AppService) {}
+
+  ngOnInit() {
+    console.log("metadata", this.metadata);
```
Comment (ericleib) [inline projects/vanilla/src/app/configurators/metadata-configurator.component.ts:36]: console.log could be removed

File: projects/vanilla/src/app/search/search-configurator.component.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+export class SearchConfiguratorComponent implements OnDestroy {
+  /**
+   * Returns the configuration of the metadata displayed in the facet-preview component.
+   * The configuration from the config.ts file can be overridden by configuration from
+   * the app configuration on the server
+   */
+  public get metadata(): string[] {
+    const m = this.appService.app?.data?.metadata as any as MetadataConfig[] || METADATA_CONFIG;
+    return m.map(i => i.field);
+  }
```
Comment (ericleib) [inline projects/vanilla/src/app/search/search-configurator.component.ts:24]: It makes sense to manage configuration like this in Vanilla Search.
But in Vanilla Builder, the configuration should always be associated to the components configuration. So there should not be a `METADATA_CONFIG` object. It should go directly in the ui-builder component's configuration object.

File: projects/vanilla/src/app/search/search.component.ts [hunk 3]
```diff
@@ -42,7 +43,14 @@ export class SearchComponent implements OnInit {
...
 
-  metadata: string[] = [];
+  /**
+   * Returns the configuration of the metadata displayed in the facet-preview component.
+   * The configuration from the config.ts file can be overridden by configuration from
+   * the app configuration on the server
+   */
+  public get metadata(): MetadataConfig[] {
+    return this.appService.app?.data?.metadata as any as MetadataConfig[] || METADATA_CONFIG;
+  }
```
Comment (ericleib) [inline projects/vanilla/src/app/search/search.component.ts:53]: Same comment here

**Review by ericleib — APPROVED** [review]

File: projects/vanilla/src/app/search/search.component.ts (hunk 3 above)
Comment (hebus) [reply projects/vanilla/src/app/search/search.component.ts:53]: I will do this. But first, to fix de CI, I'll merge this changes

**Pull request merged**